hibernate-core/src/test/java/org/hibernate/persister/entity/WhereJoinedSubclassTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.persister.entity;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Inheritance;
+import javax.persistence.InheritanceType;
+
+import org.hibernate.annotations.Persister;
+import org.hibernate.annotations.Where;
+
+import org.hibernate.testing.junit4.BaseCoreFunctionalTestCase;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+
+/**
+ * @author Daria Mitrofanova
+ */
+public class WhereJoinedSubclassTest extends BaseCoreFunctionalTestCase {
+
+	@Override
+	protected Class<?>[] getAnnotatedClasses() {
+		return new Class[] {
+				ChildEntity.class, ParentEntity.class
+		};
+	}
+
+	@Test
+	public void testSchemaNotReplacedInCustomSQL() throws Exception {
+
+		String className = ChildEntity.class.getName();
+
+		final AbstractEntityPersister persister = (AbstractEntityPersister) sessionFactory().getEntityPersister(
+				className );
+		assertEquals( persister.getSQLWhereString( "alias" ), "( alias.id = 1) " );
+		assertEquals( persister.getSqlJoinedSubclassWhereString( "alias" ), "( alias.age = 1) " );
+		assertEquals( persister.filterFragment( "alias" ), " and ( alias_1_.id = 1)  and ( alias.age = 1) " );
```

Comment:
It is a common misuse of 'assertEquals()' by replacing:
```
assertEquals( expected, actual );
```
with
```
assertEquals( actual, expected );
```
The misuse can only be noticed when assertion failed ending up with a confusing message.
```

Reply:
Thanks, it's fixed in next commit